kubernetes/samples/charts/nginx/README.md

```diff
@@ -0,0 +1,209 @@
+```
+
+#### Host-based secured routing
+This sample demonstrates how to access an application on two WebLogic domains using an HTTPS endpoint. Install a TLS-enabled ingress controller.
+
+First, you need to create two secrets with TLS certificates, one with the common name `domain1.org`, the other with the common name `domain2.org`. We use `openssl` to generate self-signed certificates for demonstration purposes. Note that the TLS secret needs to be in the same namespace as the WebLogic domain.
+```
+# create a TLS secret for domain1
+$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/tls1.key -out /tmp/tls1.crt -subj "/CN=domain1.org"
+$ kubectl -n weblogic-domain1 create secret tls domain1-tls-cert --key /tmp/tls1.key --cert /tmp/tls1.crt
+
+# create a TLS secret for domain2
+$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/tls2.key -out /tmp/tls2.crt -subj "/CN=domain2.org"
+$ kubectl -n weblogic-domain1 create secret tls domain2-tls-cert --key /tmp/tls2.key --cert /tmp/tls2.crt
+# Deploy the TLS ingress controller.
+$ kubectl create -f samples/tls.yaml
+ingress.networking.k8s.io/domain1-ingress-tls created
+ingress.networking.k8s.io/domain2-ingress-tls created
+```
+Now you can access the application on the WebLogic domain with the host name in the HTTP header. The ingress controller secure port can be obtained dynamically from the `nginx-operator` service in the `nginx` namespace.
+
+```
+# Get the ingress controller secure web port
+$ export TLS_PORT=$(kubectl -n nginx get service nginx-operator-ingress-nginx-controller -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+$ curl -k -H 'host: domain1.org' https://${HOSTNAME}:${TLS_PORT}/testwebapp/
+$ curl -k -H 'host: domain2.org' https://${HOSTNAME}:${TLS_PORT}/testwebapp/
+```
+
+## SSL termination at ingress controller
+This sample demonstrates how to terminate SSL traffic at the ingress controller to access the WebLogic Server Administration Console through the SSL port. 
+
+### 1. Enable "WebLogic Plugin Enabled" on the WebLogic domain level
+
+If you are using WDT to configure the WebLogic domain, you need to add the following resource section at the domain level to the model YAML file.
+```
+resources:
+     WebAppContainer:
+         WeblogicPluginEnabled: true
+```
+If you are using a WLST script to configure the domain, then the following modifications are needed to the respective PY script.
+```
+# Configure the Administration Server
+cd('/Servers/AdminServer')
+set('ListenPort', admin_port)
+set('Name', admin_server_name)
+set('WeblogicPluginEnabled',true)
+...
+cd('/Clusters/%s' % cluster_name)
+set('WeblogicPluginEnabled',true)
+```
+### 2. Update the Ingress resource with customRequestHeaders value.
+Replace the string 'weblogic-domain' with namespace of the WebLogic domain, the string 'domain1' with domain UID and the string 'adminserver' with name of the Administration server in the WebLogic domain.  
```

Comment:
It'd be helpful to remind folks at this point where the original resource file is located, and please also include the kubectl command for deploying the updated file below.

Reply:
The file 'nginx-tls-console.yaml' does not exist in the repository. Need to create it after copying from the doc.

Reply:
The point is that the directions here say "Update the Ingress resource " but the directions don't specify which ingress resource.  E.g. The sentences can start with "Starting with the resource file from TBD, create an updated file named 'nginx-tls-console.yaml' by replacing the string 'domain1' with ..."

Reply:
Modified the header as follows 
2. Create a ingress resource file with custom annotation value
Save the below configuration as 'nginx-tls-console.yaml' by replacing the string 'weblogic-domain' with namespace of the WebLogic domain, the string 'domain1' with domain UID and the string 'adminserver' with name of the Administration server in the WebLogic domain.

3. Deploy the ingress resource
Deploy the ingress resource using 'kubectl'.